models/fates/R/model2netcdf.FATES.R

```diff
@@ -10,581 +10,153 @@
 ##' @name model2netcdf.FATES
 ##' @title Code to convert FATES netcdf output into into CF standard
 ##'
-##' @param outdir Location of FATES model output
+##' @param outdir Location of FATES model output (e.g. a path to a single ensemble output)
+##' @param sitelat Latitude of the site
+##' @param sitelon Longitude of the site
+##' @param start_date Start time of the simulation
+##' @param end_date End time of the simulation
+##' @param pfts a named vector of PFT numbers where the names are PFT names
+##' @param settings pecan settings object
+##' @param process_partial should failed runs be processed? Defaults to `FALSE`.
+##'   `TRUE` will generate .nc files for runs that have generated some, but not
+##'   all, of the expected outputs
+##'
 ##' 
 ##' @examples  
 ##' \dontrun{
 ##' example.output <- system.file("case.clm2.h0.2004-01-01-00000.nc",package="PEcAn.FATES")
 ##' model2netcdf.FATES(outdir="~/")
 ##' }
+##'
+##' @author Michael Dietze, Shawn Serbin, Rob Kooper, Toni Viskari, Istem Fer
+## modified M. Dietze 07/08/12 modified S. Serbin 05/06/13
+## refactored by Istem Fer on 03/2018
+## further modified by S. Serbin 09/2018
 ##' 
 ##' @export
-##'
-##' @author Michael Dietze, Shawn Serbin
-model2netcdf.FATES <- function(outdir) {
 
-    # E.g. var_update("AR","AutoResp","kgC m-2 s-1", "Autotrophic Respiration")
-    # currently only works for xyt variables, need to expand to work for cohort-level outputs, 
-    # age bins, soils, etc
-    var_update <- function(out,oldname,newname,newunits=NULL,long_name=NULL){
-      if (oldname %in% ncin_names) {
-        ## define variable
-        oldunits <- ncdf4::ncatt_get(ncin,oldname,"units")$value
-        if (oldunits=="gC/m^2/s") oldunits <- "gC m-2 s-1"
-        if (oldname=="TLAI" && oldunits=="none") oldunits <- "m2 m-2"
-        if(is.null(newunits)) newunits = oldunits
-        newvar <- ncdf4::ncvar_def(name = newname, units = newunits, longname=long_name, dim = xyt)
-        
-        ## convert data
-        dat <- ncdf4::ncvar_get(ncin,oldname)
-        dat.new <- PEcAn.utils::misc.convert(dat,oldunits,newunits)
-        
-        ## prep for writing
-        if(is.null(out)) {
-          out <- list(var <- list(),dat <- list())
-          out$var[[1]] <- newvar
-          out$dat[[1]] <- dat.new
-        } else {
-          i <- length(out$var) + 1
-          out$var[[i]] <- newvar
-          out$dat[[i]] <- dat.new
-        }
+library(ncdf4.helpers) # dims names
```

Comment:
Same as `ncdf4`, no library calls in the R scripts, please remove it from here, but add it to the DESCRIPTION files